.github/workflows/publish-packages.yaml

```diff
@@ -0,0 +1,64 @@
+name: Publish
+on:
+  # NOTE: avoid using the manual execution unless is the only way to fix some issue,
+  # rather retry failed jobs in case of flakes. The manual execution can potentially
+  # cause the tags to point to a different commit that the one used to publish
+  # the packages
+  workflow_dispatch:
+  workflow_run:
+    workflows: ["Check and Test"]
+    branches: [main]
+    types:
+      - completed
+
+jobs:
+  publish:
+    if: |
+      github.event_name == 'workflow_dispatch' || (
+        startsWith(github.event.workflow_run.head_commit.message, 'chore(release): Bump package versions') &&
+        github.event.workflow_run.conclusion == 'success'
+      )
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        # don't checkout a detatched HEAD
+        ref: ${{ github.head_ref }}
+
+        # this is important so git log has the whole history
+        fetch-depth: '0'
+
+    - name: Setup git
+      run: |
+        git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git config --local user.name "github-actions[bot]"
+
+    - name: "Use NodeJS 14"
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+
+    - name: Install npm@7
+      run: npm install -g npm@7
+
+    - name: Install Dependencies
+      run: |
+        npm run bootstrap-ci
+      shell: bash
+
+    - name: "Publish what is not already in NPM"
+      env:
+        NPM_TOKEN: ${{ secrets.SVC_DEVTOOLSBOT_TOKEN }}
+      run: |
+        echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > .npmrc
+        git update-index --assume-unchanged .npmrc
+        npm run publish-packages
+
+    - name: "Publish tags"
+      run: |
+        npx lerna list -a --json | \
+          jq -r '.[] | .name + "@" + .version' | \
+          xargs -i sh -c "git tag -a {} -m {} || true"
+        git push --follow-tags
```

Comment:
I wonder if github bot will actually have enough rights to be able to push to main 🤔 Maybe we need to use our devtools bot token here?

Reply:
Works for devtools-shared

Reply:
Yeah, I think github does something special to git operations coming from inside actions – it’s a bit magical but also makes sense in a way